sdk/formrecognizer/ai-form-recognizer/src/documentModelAdministrationClient.ts

```diff
@@ -0,0 +1,613 @@
+      {
+        modelId: destinationModelId,
+        description: options.description,
+      },
+      options
+    );
+  }
+
+  /**
+   * Copies a model with the given ID into the resource and model ID encoded by a given copy authorization.
+   *
+   * See {@link CopyAuthorization} and {@link getCopyAuthorization}.
+   *
+   * ### Example
+   *
+   * ```javascript
+   * // We need a client for the source model's resource
+   * const sourceEndpoint = "https://<source resource name>.cognitiveservices.azure.com";
+   * const sourceCredential = new AzureKeyCredential("<source api key>");
+   * const sourceClient = new DocumentModelAdministrationClient(sourceEndpoint, sourceCredential);
+   *
+   * // We create the copy authorization using a client authenticated with the destination resource. Note that these two
+   * // resources can be the same (you can copy a model to a new ID in the same resource).
+   * const copyAuthorization = await client.getCopyAuthorization("<destination model ID>");
+   *
+   * // Finally, use the _source_ client to copy the model and await the copy operation
+   * const poller = await sourceClient.beginCopyModel("<source model ID>");
+   *
+   * // Model copying, like all other model creation operations, returns a poller that eventually produces a ModelInfo
+   * // object
+   * const modelInfo = await poller.pollUntilDone();
+   *
+   * const {
+   *   modelId, // identical to the modelId given when creating the copy authorization
+   *   description, // identical to the description given when creating the copy authorization
+   *   createdDateTime, // the Date (timestamp) that the model was created
+   *   docTypes // information about the document types of the model (identical to the original, source model)
+   * } = modelInfo;
+   * ```
+   *
+   * @param sourceModelId - the unique ID of the source model that will be copied
+   * @param authorization - an authorization to copy the model, created using the {@link getCopyAuthorization}
+   * @param options - optional settings for
+   * @returns a long-running operation (poller) that will eventually produce the copied model information or an error
+   */
+  public async beginCopyModel(
+    sourceModelId: string,
+    authorization: CopyAuthorization,
+    options: CopyModelOptions = {}
+  ): Promise<TrainingPoller> {
+    return this.createTrainingPoller({
+      options,
+      start: () => this._restClient.copyDocumentModelTo(sourceModelId, authorization, options),
+    });
+  }
+
+  /**
+   * Create an LRO poller that handles training operations.
+   *
+   * This is the meat of all training polling operations.
+   *
+   * @internal
+   * @hidden
+   * @param definition - operation definition (start operation method, request options)
+   * @returns a training poller that produces a ModelInfo
+   */
+  private async createTrainingPoller(
+    definition: TrainingOperationDefinition
+  ): Promise<TrainingPoller> {
```

Comment:
this could be abstracted into a standalone function and used in both clients, right?

Reply:
The `lro` function already pulls out a lot of the common abstraction. I think the current setup gives it a good balance between abstraction and readability. The only thing left that I can conceive of abstracting away is the deserialization of `resumeFrom`.

Reply:
It is fine,  I should have added a nit tag to my comment.